js/interface.js

```diff
@@ -874,150 +874,58 @@ $(window).on('resize', windowResized).trigger('resize');
 $('#app')
   .on('click', '[data-order-date]', function() {
     var $dataSource = $(this);
-    var orderedDataSources;
+    var isOrderedByDateAsc = $dataSource.hasClass('asc');
```

Comment:
I understand the logic here but we've got the same logic applied 5 times. Can this be modularised to keep it DRY?

Reply:
@tonytlwu i added a new function to make it more clear